### PR TITLE
Update sample to PHP 8.0

### PR DESCRIPTION
**Pull request by Alex-DA**

### CODE REVIEW

File: sample/README.md [hunk 1]
```diff
@@ -26,9 +26,9 @@ Perform Static Analysis:
 
 We are using this tools to add Static Analysis and Generics to our PHP code:
 
-- [PHPStan](https://phpstan.org/)
-- [Psalm](https://psalm.dev/docs/)
-- @todo Update version of PHP-CS-Fixer to PHP 8.0 `"friendsofphp/php-cs-fixer": "^2.16"`
+-   [PHPStan](https://phpstan.org/)
+-   [Psalm](https://psalm.dev/docs/)
+-   @todo Update version of PHP-CS-Fixer to PHP 8.0 `"friendsofphp/php-cs-fixer": "^2.16"`
```
Comment (doup) [inline sample/README.md]: I really hate this from Prettier… why on earth it adds spaces here? xD

File: sample/src/app/controllers/ProductController.php [hunk 2]
```diff
@@ -51,39 +51,21 @@ public function __construct(
...
-        );
-
-        return $this->renderView($variables, "index");
+        $variables['resultPutProductAction'] = $this->putProductAction($productOne);
+        $variables['resultGetProductAction'] = $this->getProductAction($productOne->getId());
+        $variables['resultPutAllProductAction'] = $this->putAllProductsAction($listProducts);
+        $variables['resultGetAllProductAction'] = $this->getAllProductsAction();
+        $variables['resultDeleteProductAction'] = $this->deleteProductAction($productOne->getId());
+
+        return $this->renderView($variables, 'index');
```
Comment (doup) [inline sample/src/app/controllers/ProductController.php]: I find this easier to read

File: core/src/shared/error/InvalidObjectException.php [hunk 3]
```diff
@@ -17,18 +17,12 @@ class InvalidObjectException extends InvalidArgumentException
...
     ) {
         if (empty($message)) {
-            $message =
-                "Error: Expected object instanceof '" .
-                $expected .
-                "'" .
-                ", received '" .
-                $received .
-                "'.";
+            $message = "Error: Expected object instanceof '" . $expected . "'" . ", received '" . $received . "'.";
```
Comment (doup) [inline core/src/shared/error/InvalidObjectException.php:25]: For me this is also easier to read

**Review by doup — COMMENTED** [review]
No big surprises, it's quite consistent/similar to what we get in TS. For me there are instances where the code gets easier to read. But it totally breaks the multiline parameters, which I assume you prefer. 🤷‍♂️

File: sample/README.md (hunk 1 above)
Comment (Alex-DA) [reply sample/README.md]: Agree.

File: sample/src/app/controllers/ProductController.php (hunk 2 above)
Comment (Alex-DA) [reply sample/src/app/controllers/ProductController.php]: Agree.

File: core/src/shared/error/InvalidObjectException.php (hunk 3 above)
Comment (Alex-DA) [reply core/src/shared/error/InvalidObjectException.php:25]: For me this is the best output:
```
$message = "Error: Expected object instanceof '" . $expected 
    . "'" . ", received '" . $received . "'.";
```

**Comment by Alex-DA** [comment]
> No big surprises, it's quite consistent/similar to what we get in TS. For me there are instances where the code gets easier to read. But it totally breaks the multiline parameters, which I assume you prefer.

Yes, all looks mostly good, the problem is the multiline that only works depending of the with.

**Pull request merged**